### ISSUE TITLE
Inconsistent Terrain Texture Painting Fidelity

### ISSUE THREAD

**Issue by Altoc** [Zylann/godot_heightmap_plugin#386]
**Describe the bug**
We are experiencing an issue where texture painting is giving us some strange behaviors. 

What we paint in our editor is inconsistent with what is shown rendered in-game. This behavior is strange, as it only occurs on one of our developers environments (we only have 2 developers, so 1 / 2).

Here is a screenshot of an example of the inconsistent texture drawing issue.
- Left is what was drawn in-editor, right is in-game.

![image](https://github.com/Zylann/godot_heightmap_plugin/assets/37989193/940721fe-5837-42da-b6c5-78732a7bfbc9)

These are the settings we are using in the HTerrain Node:
![image](https://github.com/Zylann/godot_heightmap_plugin/assets/37989193/f3c21d18-d899-40a0-ab15-b8626ee79fea)

Other notables / things we've tried:
- We are source controlling via git. The /addons/ folder is included in this
    - We have tried to install the plugin manually on Developer B's environment, to no avail.
- We gitignore the godot *.import files. We did some testing to see if removing that ignore path resolves the issue, to no avail.
- We confirmed our versions are the same in Godot, and even tried to zip up and use the same installer just to be sure.
- Taking our above screenshot as an example, if Developer A drew that, it would be consistent in game. But if Developer A pulls Developer B's branch, the issue is the same as if it were Developer B's environment.
- No console errors are output.

**To Reproduce**
1. Create HTerrain node (use supplied settings).
2. Draw terrain texture.
3. ???
4. Run scene, observe the inconsistency.

**Expected behavior**
We expect the terrain paint to resemble in-game what it resembles in-editor when being drawn.

**Screenshots**
![image](https://github.com/Zylann/godot_heightmap_plugin/assets/37989193/940721fe-5837-42da-b6c5-78732a7bfbc9)

**Environment**
Factor | Developer A | Developer B
:--: | :--: | :--:
OS | Windows 10 | Windows 10
GPU | AMD | Nvidia
Zylann Heightmap Version | 1.6.1 | 1.6.1
Godot Version | v3.4.stable.official.206ba70f4 | v3.4.stable.official.206ba70f4
Renderer | GLES3 | GLES3
Inconsistent Texture Drawing | NO | YES
Originally added the plugin | YES | NO

**Comment by Zylann** [comment]
It looks like the splatmap of the terrain is getting VRAM-compressed by Godot, which is not expected. Can you check if that's the case by inspecting import settings of the splatmap in terrain data folder?

**Comment by Altoc** [comment]
Thanks for the quick reply!

We see some differences in our import files there, strange!

Here is both Developer A's settings (he gets expected behavior, and Developer B's settings (unexpected behavior):

Developer A:

```
[remap]

importer="texture"
type="StreamTexture"
path="res://.import/splat.png-bb57e27cf774ca91787c897c90da658e.stex"
metadata={
"vram_texture": false
}

[deps]

source_file="res://terrain/SponTown/splat.png"
dest_files=[ "res://.import/splat.png-bb57e27cf774ca91787c897c90da658e.stex" ]

[params]

compress/mode=0
compress/lossy_quality=0.7
compress/hdr_mode=0
compress/bptc_ldr=0
compress/normal_map=0
flags/repeat=0
flags/filter=true
flags/mipmaps=false
flags/anisotropic=false
flags/srgb=0
process/fix_alpha_border=false
process/premult_alpha=false
process/HDR_as_SRGB=false
process/invert_color=false
process/normal_map_invert_y=false
stream=false
size_limit=0
detect_3d=false
svg/scale=1.0
```

Developer B:
```
[remap]

importer="texture"
type="StreamTexture"
path.s3tc="res://.import/splat.png-bb57e27cf774ca91787c897c90da658e.s3tc.stex"
path.etc2="res://.import/splat.png-bb57e27cf774ca91787c897c90da658e.etc2.stex"
metadata={
"imported_formats": [ "s3tc", "etc2" ],
"vram_texture": true
}

[deps]

source_file="res://terrain/SponTown/splat.png"
dest_files=[ "res://.import/splat.png-bb57e27cf774ca91787c897c90da658e.s3tc.stex", "res://.import/splat.png-bb57e27cf774ca91787c897c90da658e.etc2.stex" ]

[params]

compress/mode=2
compress/lossy_quality=0.7
compress/hdr_mode=0
compress/bptc_ldr=0
compress/normal_map=0
flags/repeat=true
flags/filter=true
flags/mipmaps=true
flags/anisotropic=false
flags/srgb=2
process/fix_alpha_border=true
process/premult_alpha=false
process/HDR_as_SRGB=false
process/invert_color=false
process/normal_map_invert_y=false
stream=false
size_limit=0
detect_3d=false
svg/scale=1.0
```

**Comment by Zylann** [comment]
Developper B somehow got wrong import settings.
The plugin writes `compress/mode=0`: https://github.com/Zylann/godot_heightmap_plugin/blob/5d447445c21b1afe7f51b7987485c54854554af0/addons/zylann.hterrain/hterrain_data.gd#L1215
But developper B somehow has `compress/mode=2` which means VRAM compressed. I don't know why Godot changes this, but you can fix the issue by switching compression off again.

**Comment by Altoc** [comment]
Initial testing on Developer B using Developer A's import settings shows the issue as being resolved!

Interesting, it sounds like the chain of events was:

1. Developer A installed the plugin, the correct import settings came with the plugin (?)
2. Developer B pulled the plugin from source, the incorrect, default import settings were initialized as the imports were not version controlled

We both admire your prompt response time, huge kudos to you. I believe our issue has been resolved. We will look into removing the ignore for the import files, it has worked well enough up until plugins came into the picture, and I am not really sure why we started ignoring them in the first place anyway.

And another thank you for the solution on resolving our import issue via git. I shall do as you suggest, and I believe this issue can be closed out.
